src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -263,6 +272,60 @@ public BulkInsertOperation(string database, IDocumentStore store, BulkInsertOpti
 
             _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(_requestExecutor.Conventions,
                 entity => AsyncHelpers.RunSync(() => _requestExecutor.Conventions.GenerateDocumentIdAsync(database, entity)));
+
+            _streamLock = new AsyncReaderWriterLock();
+            _lastWriteToStream = DateTime.UtcNow;
+            _timer = new Timer(HandleHeartbeat,
+                new WeakReference<BulkInsertOperation>(this),
+                HeartbeatCheckInterval,
+                HeartbeatCheckInterval);
+        }
+
+        private static void HandleHeartbeat(object state)
+        {
+            var weakReference = (WeakReference<BulkInsertOperation>)state;
+            if (weakReference.TryGetTarget(out BulkInsertOperation bulkInsert) == false)
+                return;
```

Comment:
In this case, we need to cancel the timer. 
The state object you get should also include the timer, so you can dispose it in this case, no?